bindings/angular2/src/directives/ons-input.ts

```diff
@@ -77,4 +88,14 @@ export class OnsInput implements OnChanges, OnDestroy {
 
     this._element = null;
   }
+
+  writeValue(obj: any) {
+    this._element.value = obj;
+  }
+ 
+  registerOnChange(fn: any) {
+     this._propagateChange = fn;
+  }
+ 
+  registerOnTouched() { }
```

Comment:
Is this empty `registerOnTouched` function actually required?

Reply:
Yes, it is required by `ControlValueAccessor ` interface.
https://angular.io/api/forms/ControlValueAccessor

Reply:
Thanks for the link to the docs. If we do not call the argument `fn` as described in the docs, does Angular still mark the fields as touched correctly?

Reply:
Unfortunately no.
I tried two examples but `touched` property didn't change to `true`.

```html
<form [formGroup]="form">
  <ons-input formControlName="value"></ons-input>
</form>
{{form.controls['value']?.touched}}  <!-- false -->
```

```html
<form #f="ngForm">
  <ons-input name="value" [(ngModel)]="value"></ons-input>
</form>
{{f.form.touched}}  <!-- false -->
```

Reply:
If we completely remove `registerOnTouched`, does it throw an error? My expectation would be that if it doesn't exist, Angular would use the default behaviour.

If removing `registerOnTouched` doesn't work, perhaps we can change it to:
```
registerOnTouched(fn: any) {
  fn();
}
```

Reply:
Removing `registerOnTouched` causes compile error. This is because of TypeScript specification.

I think this may be better.

```ts
private _propagateTouched = () => {};

@HostListener('blur')
_onBlur() {
  this._propagateTouched();
}

registerOnTouched(fn: any) {
  this. _propagateTouched = fn;
}
```

Reply:
Ah yes, of course. My suggestion would not work, because it was called in the registration handler. This looks like a good solution to me. :+1: